librocksdb-sys/build.rs

```diff
@@ -359,13 +359,26 @@ fn main() {
         update_submodules();
     }
     bindgen_rocksdb();
+    let target = env::var("TARGET").unwrap();
 
     if !try_to_find_and_link_lib("ROCKSDB") {
+        // rocksdb only works with the prebuilt rocksdb system lib on freebsd.
+        // we dont need to rebuild rocksdb
+        if target.contains("freebsd") {
+            println!("cargo:rustc-link-search=native=/usr/local/lib");
+            let mode = match env::var_os("ROCKSDB_STATIC") {
+                Some(_) => "static",
+                None => "dylib",
+            };
+            println!("cargo:rustc-link-lib={}=rocksdb", mode);
+
+            return;
```

Comment:
given that we return here, we won't try to build snappy https://github.com/rust-rocksdb/rust-rocksdb/pull/908/files#diff-7377ccc9f5cb3386318bd8afcd84814e7dd6d9b40efa909fb2dc218ecb779499R389 is this expected?

Reply:
Oops probably not. Though I did see snappy in their `ldd` output. I can still move it around and ask them to try building again, but we do need to return somewhere before the `build_rocksdb();`.

Reply:
Ya, check here: https://www.freshports.org/databases/rocksdb/

On Library dependencies section, snappy is listed and rocksdb won't install without it